sdk/src/asset_handlers/riff_io.rs

```diff
@@ -298,12 +297,10 @@ impl CAIReader for RiffIO {
             reader: input_stream,
         };
 
-        for c in top_level_chunks.iter(&mut chunk_reader) {
-            if c.id() == XMP_CHUNK_ID {
-                let output = c.read_contents(&mut chunk_reader).ok()?;
-                let output_string = String::from_utf8_lossy(&output);
-
-                return Some(output_string.to_string());
+        for chunk in top_level_chunks.iter(&mut chunk_reader).flatten() {
```

Comment:
This implementation is skipping chunks that have an error and continuing with the iterator. We could change this to return a None after first finding an invalid chunk. Any thoughts?

Reply:
Yeah, I would fail on invalid content, that is the issue with **flatten**.  You will skip problems.  I think there was another flatten in the changes.

Reply:
Will change, thanks!

Reply:
Is it common for RIFF to have invalid content that we can skip and the file remain useful?

Reply:
It appears to depend on the implementation. Apple Music and Premiere Pro were fine with the incorrect length header in the DATA chunk. Not sure how they are reading the audio data. The data chunk was the last chunk in this example. Maybe they are checking for the end of the file as a safe guard?